lib/Dav/Faces/FacePhoto.php

```diff
@@ -51,12 +51,14 @@ public function getName() {
 	 */
 	public function delete() {
 		$detections = $this->detectionMapper->findByClusterId($this->faceDetection->getClusterId());
-		$centroid = FaceClusterAnalyzer::calculateCentroidOfDetections($detections);
-		$distance = new Euclidean();
-		$distanceValue = $distance->compute($centroid, $this->faceDetection->getVector());
+		if (count($detections) > 1) {
+			$centroid = FaceClusterAnalyzer::calculateCentroidOfDetections($detections);
+			$distance = new Euclidean();
+			$distanceValue = $distance->compute($centroid, $this->faceDetection->getVector());
+			// Set threshold to avoid recreating the same mistake
+			$this->faceDetection->setThreshold($distanceValue);
```

Comment:
Here we only want to calculate and set a threshold if there's more than the one face in the cluster, otherwise we do a useless calculation.